### PR TITLE
Sync public repo 16

### PR DESCRIPTION
**Pull request by jthompson6**
@rachelwarren @taggartgorman merge?

### CODE REVIEW

File: plugin-core/src/main/scala/com/alpine/plugin/core/OperatorParameters.scala [hunk 1]
```diff
@@ -89,4 +89,5 @@ trait OperatorParameters {
    */
   def getParameterIds: Iterator[String]
 
+  def getAdvancedSparkParameters: scala.collection.mutable.Map[String, String]
```
Comment (jthompson6) [inline plugin-core/src/main/scala/com/alpine/plugin/core/OperatorParameters.scala]: I am wondering if we should make this an immutable map.

Comment (taggartgorman) [inline plugin-core/src/main/scala/com/alpine/plugin/core/OperatorParameters.scala]: I do not recall there being any code that depends on this being mutable. I can change it.

Comment (taggartgorman) [inline plugin-core/src/main/scala/com/alpine/plugin/core/OperatorParameters.scala]: Switching this over to immutable would result in changes in 6 files. I need to take a deeper look first.

Comment (jthompson6) [inline plugin-core/src/main/scala/com/alpine/plugin/core/OperatorParameters.scala]: That's fine. I'll merge it as is for this alpha version, and we can finalise later.

**Comment by jthompson6** [comment]
LGTM

**Pull request merged**